documentTemplates.list - documentation

JordyDeGroeve opened the pull request.

JordyDeGroeve committed:
Add documentation for the templates.list endpoint

JordyDeGroeve committed:
Add the newly created file to the apiary file

mrtus commented on apiary.apib:
```diff
@@ -762,6 +762,39 @@ Get a list of all work types, sorted alphabetically (on their name).
+                    + order
+                    + order_confirmation
+                    + quotation
+                    + timetracking_report
```
Q: Should we convert this to `time_tracking` instead? Or are there more time tracking templates?

JordyDeGroeve replied:
This refers to a report containing all the info about multiple trackings.
If we'd change this I' d rather go for `timesheet`. But we'll leave it like it is.

jenssegers replied:
Yes, let's change this!

jenssegers replied:
Any updates on this one?

dietrichm replied:
Let's keep it as is for the time being. We have no margin in our sprint for renaming this stuff. But feel free to take initiative if you want.

jenssegers replied:
Can we remove it from the documentation in the meanwhile then? I would rather publish it without `timetracking_report` then having to do a BC-break later.

JordyDeGroeve replied:
And support all but one document template?

dietrichm replied:
Since we use it internally only (even just for WO) we decided to merge on the WO docs integration branch and do the necessary before releasing that one.

mrtus commented on apiary.apib:
```diff
@@ -762,6 +762,39 @@ Get a list of all work types, sorted alphabetically (on their name).
+                    + type: `department` (string)
+                    + id: `36386b05-936e-4cc0-9523-bd20d797ebf5` (string)
+                + type: `invoice` (string)
+                + is_default: true (boolean)
```
Q: `is_default` should we remove the `is` making it `default`? looking at the invoicing endpoint we use `paid` and not `is_paid` we also use `invoicable` instead of `is_invoicable`

JordyDeGroeve replied:
Valid point. @dietrichm, @jenssegers Should/can I change this?

rathesDot replied:
I don't agree. For Paid, you can ask `Paid? Yes/No` ... Same for `Invoicable? Yes/No?` ... But if you try to do the same for `default?` you will automatically as for `is Default? Yes/No`

JordyDeGroeve replied:
Indeed, I assume if I see a `default` key, I'll get an identifier or object that is the default.
But I get what @mrtus is saying.

dietrichm replied:
For me it can stay as `is_default`.

mrtus replied:
Is it paid, is it invoicable, is it default? That is what I would ask @rathesDot but sure if we decide this is correct for this use-case we can leave as is.

jenssegers replied:
I was also a bit doubting about the `is_` to be honest. I'm also ok with just having `default`, but no real preference to one.

JordyDeGroeve replied:
Okay, I'll leave it `is_default`.

dietrichm commented on apiary.apib:
```diff
@@ -762,6 +762,39 @@ Get a list of all work types, sorted alphabetically (on their name).
 
+## Templates [/templates]
+
+### templates.list [POST /templates.list]
```
B: endpoints retrieving data should not be documented as using `POST`
```suggestion
### templates.list [GET /templates.list]
```

dietrichm commented on apiary.apib:
```diff
@@ -762,6 +762,39 @@ Get a list of all work types, sorted alphabetically (on their name).
                 + name: `Planning` (string)
 
+## Templates [/templates]
+
```
PP: denote that this is about document layout templates, especially given that this is not within some more specific section in the docs.

JordyDeGroeve replied:
Any idea how this can be described as clear as possible?

dietrichm commented on apiary.apib:
```diff
@@ -762,6 +762,39 @@ Get a list of all work types, sorted alphabetically (on their name).
+    + Attributes (object)
+        + filter (object, optional)
+            + department_id: `a344c251-2494-0013-b433-ccee8e8435e5` (string)
+            + type: `invoice` (enum[string])
```
PP: add `optional` to both optional filters.

JordyDeGroeve replied:
Doublechecking with Charlotte if these are optional. Otherwise, I'll need to add the required param

JordyDeGroeve replied:
Changes are implemented

dietrichm commented on apiary.apib:
```diff
@@ -762,6 +762,39 @@ Get a list of all work types, sorted alphabetically (on their name).
+                + department (object)
+                    + type: `department` (string)
+                    + id: `36386b05-936e-4cc0-9523-bd20d797ebf5` (string)
+                + type: `invoice` (string)
```
PP: should be `enum[string]` as the filter param. Also include the possible members.

JordyDeGroeve committed:
read endpoints should use GET

JordyDeGroeve committed:
The returned type is also an enum string.

JordyDeGroeve committed:
Rebuild apiary file

dietrichm commented:
Hold until checked by @jenssegers

jenssegers commented on apiary.apib:
```diff
@@ -762,6 +762,47 @@ Get a list of all work types, sorted alphabetically (on their name).
                 + id: `811a5825-96f4-4318-83c3-2840935c6003` (string)
                 + name: `Planning` (string)
 
+## Templates [/templates]
```
```suggestion
## Document Templates [/documentTemplates]
```

jenssegers commented on apiary.apib:
```diff
@@ -762,6 +762,47 @@ Get a list of all work types, sorted alphabetically (on their name).
 
+## Templates [/templates]
+
+### templates.list [GET /templates.list]
```
```suggestion
### documentTemplates.list [GET /documentTemplates.list]
```

jenssegers commented on apiary.apib:
```diff
@@ -762,6 +762,47 @@ Get a list of all work types, sorted alphabetically (on their name).
+    + Attributes (object)
+        + filter (object, optional)
+            + department_id: `a344c251-2494-0013-b433-ccee8e8435e5` (string)
+            + type: `invoice` (enum[string])
```
```suggestion
            + document_type: `invoice` (enum[string])
```

jenssegers commented on apiary.apib:
```diff
@@ -762,6 +762,47 @@ Get a list of all work types, sorted alphabetically (on their name).
+                + department (object)
+                    + type: `department` (string)
+                    + id: `36386b05-936e-4cc0-9523-bd20d797ebf5` (string)
+                + type: `invoice` (enum[string])
```
```suggestion
                + document_type: `invoice` (enum[string])
```

JordyDeGroeve committed:
Make both filters required

There was some unclarity about the requirements for filters.
By making it required now, making it optional later will not be a BC

JordyDeGroeve committed:
Rename key for clarity about the type

JordyDeGroeve committed:
Build the apiary file

mrtus reviewed (dismissed):
`/templates` should become `/documentTemplates`

JordyDeGroeve committed:
Rename the endpoint for extra clarity

JordyDeGroeve commented:
Damn, committing and pushing is hard 🙃 😫
thanks for checking @mrtus

dietrichm commented:
✅ Checked

jenssegers commented on apiary.apib:
```diff
@@ -762,6 +762,47 @@ Get a list of all work types, sorted alphabetically (on their name).
+                    + order
+                    + order_confirmation
+                    + quotation
+                    + timetracking_report
```
```suggestion
```

jenssegers commented on apiary.apib:
```diff
@@ -762,6 +762,47 @@ Get a list of all work types, sorted alphabetically (on their name).
+                        + order
+                        + order_confirmation
+                        + quotation
+                        + timetracking_report
```
```suggestion
```

JordyDeGroeve commented:
@jenssegers We will reevaluate your remark to rename/leave out the `timetracking_report` once we are merging the base branch to master.
